File: server/Infrastructure/LobTools/Models/ReportModels.cs
namespace Brainvest.Dscribe.LobTools.Models
{
	public class ReportsListResponse
	{
		public string EntityTypeName { get; set; }
		public ReportFormats Format { get; set; }
		public int Id { get; set; }
		public string Title { get; set; }
	}

	public class DownloadReportRequest
	{
		public int ReportDefinitionId { get; set; }
		public int EntityIdentifier { get; set; }
	}

	public class SaveReportAsAttachmentRequest
	{
		public int ReportDefinitionId { get; set; }
		public int EntityIdentifier { get; set; }
		public string Title { get; set; }
		public string Description { get; set; }
	}
}

File: client/src/lib/lob-tools/models/report-models.ts
export enum ReportFormats {
	RichTextDocument = 1
}

export class ReportsListResponse {
	EntityTypeName: string;
	Format: ReportFormats;
	Id: number;
	Title: string;
}

export class DownloadReportRequest {
	ReportDefinitionId: number;
	EntityIdentifier: number;
}

export class SaveReportAsAttachmentRequest {
	ReportDefinitionId: number;
	EntityIdentifier: number;
	Title: string;
	Description: string;
}
